### ISSUE TITLE
Emoji converter doesn't work with custom emojis

### ISSUE THREAD

**Issue by zt64** [Kotlin-Discord/kord-extensions#190]
# Description

Emoji converter throws not found error for all custom emojis.

![image](https://user-images.githubusercontent.com/31907977/192053795-cd65dfeb-6e8f-4eba-a736-bb25d4b6f5a7.png)

# Versions

Kord Extensions: 1.5.5-SNAPSHOT

# Reproduction

```kt
val emoji by emoji {
    name = "emoji"
    description = "The emoji"
}
```

**Comment by gdude2002** [comment]
The code appears as if it should work as intended, but I can see Kord's caching potentially getting in the way. Still, please check the following:

* That you have the necessary intents to see guild emoji
* That you're using an emoji that exists on a guild the bot is present on
* If that doesn't help, that the bot has permission to manage emojis on the server

If none of these things help, report back and let me know.

Additionally, it's possible that this converter isn't actually all that useful in the long run - maybe it should return some other type? I'm not sure how best to change it, because you need a guild ID in order to grab a `GuildEmoji` object. It's possible that Kord hasn't implemented the logic to make this kind of converter realistic.

**Comment by zt64** [comment]
Turns out adding the `Intent.GuildEmojis` fixed it. Maybe there should be documentation added to the emoji that says it will need that intent.

**Comment by gdude2002** [comment]
This has been documented in [the new KordEx docs](https://docs.kordex.dev/converters.html#bundled). I've also added a note to the KDoc for the emoji converter, although I realise it's unlikely anyone actually reads those.

As an update on this converter in general, I figured I'd also mention that the emoji converter can now handle Unicode emoji names as defined by Discord, and KordEx now provides [JEmoji](https://github.com/felldo/JEmoji) as an API dependency for this purpose.
